css/cssom-view/scrollIntoView-horizontal-tb-writing-mode-and-rtl-direction.html

```diff
@@ -0,0 +1,122 @@
+<!DOCTYPE html>
+<title>CSSOM View - scrollIntoView considers horizontal-tb and rtl direction</title>
+<meta charset="utf-8">
+<link rel="author" title="Cathie Chen" href="mailto:cathiechen@igalia.com">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#dom-element-scrollintoview">
+<link rel="help" href="https://drafts.csswg.org/cssom-view/#scroll-an-element">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<style>
+.box {
+  float: left;
+  width: 200px;
+  height: 200px;
+}
+#scroller {
+  direction: rtl;
+  overflow: scroll;
+  width: 300px;
+  height: 300px;
+}
+#container{
+  width: 600px;
+  height: 600px;
+}
+#target {
+  background-color: #ff0;
+}
+</style>
+<body>
+<div id="scroller">
+  <div id="container">
+    <!--  ROW-1  -->
+    <div class="row">
+      <div class="box"></div>
+      <div class="box"></div>
+      <div class="box"></div>
+    </div>
+
+    <!--  ROW-2  -->
+    <div class="row">
+      <div class="box"></div>
+      <div class="box" id="target"></div>
+      <div class="box"></div>
+    </div>
+
+    <!--  ROW-3  -->
+    <div class="row">
+      <div class="box"></div>
+      <div class="box"></div>
+      <div class="box"></div>
+    </div>
+  </div>
+</div>
+
+<script>
+// In horizontal-tb mode and rtl direction, X corresponds to the inline axis
+// and is oriented leftward. Y corresponds to the block axis and is oriented
+// downward. So the beginning edges are the top and right edges and the ending
+// edges are the bottom and left edges.
+
+// According to the spec, x is min(0, max(x, element padding edge width - element scrolling area width)).
+// So x is nonpositive and decreases leftward.
+
+// This assumes that the horizontal scrollbar is on the bottom side
+// and the vertical scrollbar is on the left side.
+
+var target = document.getElementById("target");
+var scroller = document.getElementById("scroller");
+var scrollbar_width = scroller.offsetWidth - scroller.clientWidth;
+
+var scroller_width = scroller.offsetWidth;
+var scroller_height = scroller.offsetHeight;
+var box_width = target.offsetWidth;
+var box_height = target.offsetHeight;
+
+var expectedX = {
+  inlineStart: -box_width,
+  inlineCenter: -((3*box_width - scroller_width)/2) - scrollbar_width/2,
+  inlineEnd: -(2*box_width - scroller_width) - scrollbar_width,
+};
+
+var expectedY = {
+    blockStart: box_height,
+    blockCenter: (3*box_height - scroller_height)/2 + scrollbar_width/2,
+    blockEnd: 2*box_height - scroller_height + scrollbar_width,
+  };
+
+// This formats dict as a string suitable as test name.
+// format_value() is provided by testharness.js,
+// which also preserves sign for -0.
```

Comment:
I don't understand the comment about -0 ; the values are all strings.
It seems the objects are simple enough and don't require to introduce a new format_dict function.